### PR TITLE
Add support for JSON logging

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
Signed-off-by: Waleed Malik <ahmedwaleedmalik@gmail.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Add support for JSON logging. New flag `--log-format` was introduced to specify the format to use for logging. Currently, JSON and text are supported, with text being the default.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
--log-format flag is introduced to choose between text and JSON formatted logging. 
```

### CODE REVIEW

File: pkg/cmd/shared.go [hunk 1]
```diff
@@ -133,14 +134,28 @@ func persistentGlobalOptions(fs *pflag.FlagSet) (*globalOptions, error) {
...
+func newLogger(verbose bool, format string) *logrus.Logger {
 	logger := logrus.New()
-	logger.Formatter = &logrus.TextFormatter{
-		FullTimestamp:   true,
-		TimestampFormat: "15:04:05 MST",
+
+	switch format {
+	case "json":
+		logger.Formatter = &logrus.JSONFormatter{
+			TimestampFormat: "15:04:05 MST",
```
[the remark this answers is not in the document]
Comment (ahmedwaleedmalik) [reply pkg/cmd/shared.go:152]: I just copied it over from the `text` format settings. But IMO we should just use the default, that is 2006-01-02T15:04:05Z07:00 (RFC3339).

[the remark this answers is not in the document]
Comment (xmudrii) [reply pkg/cmd/shared.go:152]: RFC3339 sounds fine as well.

**Comment by xmudrii** [comment]
/lgtm
will leave approval for @kron4eg

**Pull request merged**